string name
time stamp
byte[] mask